src/obr/cli.py

```diff
@@ -79,6 +79,23 @@ def is_valid_workspace(filters: list = []) -> bool:
     return True
 
 
+def cli_cmd_setup(kwargs: dict) -> tuple[OpenFOAMProject, Job]:
+    """This function performs the common pattern of checking project folders for existence and creating the project and extracting the jobs."""
+    if kwargs.get("folder"):
+        os.chdir(kwargs["folder"])
+    project = OpenFOAMProject.get_project()
+    filters: list[str] = kwargs.get("filter", [])
+    if sel := kwargs.get("job"):
+        jobs = [job for job in project if sel == job.id]
+    else:
+        jobs = project.filter_jobs(filters=filters)
```

Comment:
I think we could implement `--job` and `--filter` as mututally excluse. It would introduce additional dependencies though: https://github.com/click-contrib/click-option-group